cut_flowline_module.py

```diff
@@ -0,0 +1,136 @@
+from PyQt5.QtCore import QVariant
+from qgis._core import QgsWkbTypes, Qgis, QgsVectorLayer, QgsField, QgsProject, QgsFeature
+from qgis._gui import QgisInterface
+from .utils import LayerUtils
+
+
+def delete_points_outside_range(min_id, max_id, point_layer):
+    """
+    Deletes points outside the selected ID range.
+
+    Args:
+        min_id (int): Minimum ID in the range
+        max_id (int): Maximum ID in the range
+        point_layer (QgsVectorLayer): The layer containing the points
+    """
+    point_layer.startEditing()
+    for feature in point_layer.getFeatures():
+        point_id = feature["id"]
```

Comment:
What if the feature doesn't have an `id` field?
Example solution: 
```suggestion
        if selected_layer.fields().indexFromName(self.qgis_field_name) == -1:
            self.iface.messageBar().pushMessage(
                "Please select a vector layer with an id field",
                level=Qgis.Warning,
                duration=4,
            )
            return            
        point_id = feature["id"]
```